drizzlepac/outputimage.py

```diff
@@ -744,20 +766,25 @@ def deleteDistortionKeywords(hdr):
     """ Delete distortion related keywords from output drizzle science header
         since the drizzled image should have no remaining distortion.
     """
-    dist_kws = ['D2IMERR1','D2IMERR2','D2IMDIS1','D2IMDIS2','D2IM1.*','D2IM2.*',
-                'D2IMEXT', 'DP1', 'DP2', 'CPDIS?', 'CPERR?']
+    dist_kws = ['D2IMERR1', 'D2IMERR2', 'D2IMDIS1', 'D2IMDIS2',
+                'D2IMEXT', 'DP1', 'DP2', ]
+    dist_multi_kws = ['D2IM1.*', 'D2IM2.*', 'CPDIS?', 'CPERR?']
     for kw in dist_kws:
         if kw in hdr:
             del hdr[kw]
+    for multi_kw in dist_multi_kws:
+        for kw in hdr[multi_kw]:
+            if kw in hdr:
+                del hdr[kw]
```

Comment:
```suggestion
    for multi_kw in dist_multi_kws:
        del hdr[multi_kw]
```
This works just fine.

Reply:
Unfortunately, if the keyword does not exist, the line 'del hdr[multi_kw]' will actually cause an Exception to be thrown, which is what my version explicitly avoids.

Reply:
Have you tried this? Here is my test:
```python
In [2]: from astropy.io import fits
   ...: import numpy as np
   ...: fo = fits.HDUList()
   ...: hdr = fits.Header()
   ...: hdr['TELESCOP'] = 'HST'

In [3]: hdr
Out[3]: TELESCOP= 'HST     '

In [4]: del hdr['test*']

In [5]:
```

Reply:
Actually, this is what I get:

```
In [409]: hdu = fits.open('test_sregion_drc.fits')

In [410]: scihdr = hdu[1].header.copy()
In [414]: 'sregion' in scihdr
Out[414]: False

In [415]: del scihdr['sregion']
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-415-64a3a51e610a> in <module>
----> 1 del scihdr['sregion']

c:\users\warren\miniconda3\envs\rthstdev\lib\site-packages\astropy\io\fits\header.py in __delitem__(self, key)
    228                 # To delete keyword without caring if they were present,
    229                 # Header.remove(Keyword) can be used with optional argument ignore_missing as True
--> 230                 raise KeyError(f"Keyword '{key}' not found.")
    231
    232             for idx in reversed(indices[key]):

KeyError: "Keyword 'SREGION' not found."

In [416]: del scihdr['sregion*']

```
Your test example worked because 'del' has different rules for deleting multiple keywords at once as opposed to deleting individual keywords.  This forces the use of '.pop' or checking for the existence of the individual specific keywords before trying to work with them.

Reply:
This has been cleaned up in 4b565a7 by using 'del' and '.pop' appropriately avoiding the need to check for the existence of the keywords ahead of time.